Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -407,73 +397,108 @@ extension NIOSSLCertificate: Hashable {
     }
 }
 
-/// A helper sequence object that enables us to represent subject alternative names
-/// as an iterable Swift sequence.
-internal class SubjectAltNameSequence: Sequence, IteratorProtocol {
-    typealias Element = NIOSSLCertificate.AlternativeName
+public struct _SubjectAlternativeName {
+    public var nameType: NameType
+    public var contents: Contents
 
-    private let nameStack: OpaquePointer
-    private var nextIdx: Int
-    private let stackSize: Int
+    public struct NameType: Hashable {
+        var rawValue: Int
 
-    init(nameStack: OpaquePointer) {
-        self.nameStack = nameStack
-        self.stackSize = CNIOBoringSSLShims_sk_GENERAL_NAME_num(nameStack)
-        self.nextIdx = 0
+        fileprivate init(_ rawCode: Int) {
```

Comment:
This, and `var rawValue`, may as well be `public`: the API is more useful if users can refer to other types.